lib/beacon_api/utils.ex

```diff
@@ -10,4 +10,20 @@ defmodule BeaconApi.Utils do
   def is_bytes32?(value) when is_binary(value) do
     String.starts_with?(value, "0x") and byte_size(value) == 66
   end
+
+  @spec parse_id(binary) :: atom | binary
+  def parse_id("genesis"), do: :genesis
+  def parse_id("justified"), do: :justified
+  def parse_id("finalized"), do: :finalized
+  def parse_id("head"), do: :head
+
+  def parse_id("0x" <> hex_root) when byte_size(hex_root) == 64, do: Base.decode16!(hex_root)
```

Comment:
We should use the fallible version (the one without the `!`) and map errors to `:invalid_id`